src/codegen/sdk/typescript/import_resolution.py

```diff
@@ -153,6 +155,17 @@ def resolved_symbol(self) -> Symbol | ExternalModule | TSFile | None:
         if resolved_symbol is None:
             return None
 
+        # Track namespace-chained usages before any resolution happens
+        if hasattr(resolved_symbol, "get_symbol"):
```

Comment:
Can we not call this in resolved_symbol - it should be part of the compute dependencies process or it's own separate stage like superclasses